src/Workspaces/Core/Portable/Storage/SQLite/v1/Interop/SqlConnection.cs

```diff
@@ -141,10 +138,32 @@ public ResettableSqlStatement GetResettableStatement(string query)
         {
             if (!_queryToStatement.TryGetValue(query, out var statement))
             {
-                var result = (Result)raw.sqlite3_prepare_v2(_handle, query, out var rawStatement);
+                using var _ = _handle.Lease();
+
+                var result = (Result)raw.sqlite3_prepare_v2(_handle.DangerousGetHandle(), query, out var rawStatement);
                 ThrowIfNotOk(result);
-                statement = new SqlStatement(this, rawStatement);
-                _queryToStatement[query] = statement;
+
+                SafeSqliteStatementHandle wrapper;
+                try
+                {
+                    wrapper = new SafeSqliteStatementHandle(_handle, rawStatement);
+                }
+                catch
+                {
+                    raw.sqlite3_finalize(rawStatement);
+                    throw;
+                }
```

Comment:
any way to make a helper for this common pattern?  i.e. "try to run this code, if it fails, run this cleanup".  And then just use the trick of passing along static-funcs along with args to make it alloc free?

Reply:
This is not a common pattern. Normally `sqlite_prepare_v2` would provide the `SafeHandle` directly.

Reply:
I'll think about if there is a way to simplify this.

Reply:
thanks!

Reply:
This has now been simplified.